modular_skyrat/modules/nuclear_reactor/code/reactor.dm

```diff
@@ -0,0 +1,410 @@
+	process_temperature()
+
+	// Now we process the pressure.
+	process_pressure()
+
+	// Now we'll check if the above changes damage the reactor hull.
+	process_damage()
+
+	// And finally we process effects such as alarms, flames, etc.
+	process_effects()
+
+	update_appearance()
+
+	//if(core_pressure <= 0)
+		//shut_down()
+
+/**
+ * A reactor can be destroyed in the following ways:
+ *
+ * 1. The reactor has reached maximum temperature and is literally melting. This causes the reactor to meltdown.
+ * A meltdown is not as catastrophic as a blowout, but is still plenty dangerous. It will begin oozing out radioactive lava and toxic gasses,
+ * and the room will fill with fire. Eventually the reactor will detonate in a small explosion leaving a neutron beam of radiation in a random direction.
+ *
+ * 2. The reactor has reached maximum pressure and is going to explode. This causes the reactor to blowout.
+ * A blowout is far more catastrophic than a meltdown, and will cause the most damage to the station. When a reactor blowsout, it will violently explode in
+ * a very large fireball, leaving the station in a state of chaos as radioactive gasses and toxic gasses are released in all directions, landing all across the station.
+ */
+/obj/machinery/reactor/proc/process_damage()
+	if(core_pressure >= REACTOR_PRESSURE_MAXIMUM) // Pressure will kill the reactor before a meltdown happens.
+		vessel_integrity -= clamp(core_pressure / 10000, 0, REACTOR_MAX_DAMAGE_PER_SECOND)
+		if(vessel_integrity <= 0)
+			blowout()
+			return
+
+	if(core_temperature >= REACTOR_TEMPERATURE_MAXIMUM)
+		vessel_integrity -= clamp(core_temperature / 1000, 0, REACTOR_MAX_DAMAGE_PER_SECOND)
+		if(vessel_integrity <= 0)
+			meltdown()
+			return
+
+/obj/machinery/reactor/proc/meltdown()
+
+/obj/machinery/reactor/proc/blowout()
+
+/obj/machinery/reactor/proc/process_effects()
+
+/obj/machinery/reactor/proc/start_up()
+	START_PROCESSING(SSmachines, src)
+	playsound(src, pick('modular_skyrat/modules/nuclear_reactor/sound/startup2.ogg', 'modular_skyrat/modules/nuclear_reactor/sound/startup.ogg'), 100, TRUE)
+	soundloop.start()
+	visible_message(span_notice("[src] clunks loudly as it begins to whirr to life!"))
+	icon_state = "running"
+
+/obj/machinery/reactor/proc/shut_down()
+	STOP_PROCESSING(SSmachines, src)
+	soundloop.stop()
+	core_temperature = T20C
+
+/obj/machinery/reactor/proc/process_pressure() //Pressure is what drives the turbine. If the vent is closed, pressure will quickly build up.
+	var/pressure_modifier = core_pressure
+
+	if(core_temperature <= REACTOR_TEMPERATURE_MINIMUM) // We are going to drop  pressure as steam is now turning into water.
+		pressure_modifier -= REACTOR_PRESSURE_CONDENSATION_RATE * core_temperature
+	else // We are going to increase pressure as water is now turning into steam.
+		pressure_modifier += REACTOR_PRESSURE_EVAPORATION_RATE * core_temperature
+
+	core_pressure = clamp(pressure_modifier, 0, INFINITY) // We cannot go below zero pressure.
```

Comment:
negative pressure is a real thing!

Reply:
Not in this reactor, negative pressure is not something I will support